### PR TITLE
fix(UVE): unable to execute sub-action on pages

### LINKED ISSUE #29920: UVE: Unable to execute sub-actions
### Problem Statement

- [x] Users are currently unable to execute sub-actions on pages

Users are currently unable to execute sub-actions on pages within the dotCMS platform. This issue affects the functionality of sub-actions, such as copying, publishing, or archiving pages, which are critical operations for content management workflows. The inability to perform these sub-actions impacts both content creators and administrators, leading to delays in content updates and management. This issue occurs across multiple browsers and operating systems, with no discernible difference in behavior between them.

https://github.com/user-attachments/assets/f7a86c7f-61af-400b-a0e9-26f4ae8d8849

### Acceptance Criteria

- Sub-actions should provide immediate feedback to users, confirming whether the action was successful or failed.

### dotCMS Version

Tested on trunk // Docker // FF 

### Proposed Objective

Quality Assurance

### Proposed Priority

Priority 1 - Show Stopper

### PR DESCRIPTION
**Pull request by rjvelazco**
### Proposed Changes
*  Let users execute `sub-actions` on pages

### Checklist
- [x] Tests

### Why was this error happening?
It seems like an incompatibility between Angular's `SlicePipe` and PrimeNg v17

A general guess, `SlicePipe` is not a Pure Pipe and has Angular's own logic implemented inside that doesn't work like a native slice, which makes it inconsistent with what one would expect it to work like by @zJaaal 

#### Test the `Slice` pipe with PrimeNg `v15` and `v17`

- PrimeNG v15 - slice [demo](https://stackblitz.com/edit/xad6jf?file=src%2Fapp%2Fdemo%2Fsplit-button-basic-demo.ts,src%2Fapp%2Fdemo%2Fsplit-button-basic-demo.html)
- PrimeNG v17 - slice [demo](https://stackblitz.com/edit/ixjqxa?file=src%2Fapp%2Fsplit-button-basic-demo.ts)

### Video

https://github.com/user-attachments/assets/2ae68864-155c-4c53-903f-7f233d8021af

### CODE REVIEW

**Review by valentinogiardino — APPROVED** [review]

File: core-web/libs/ui/src/lib/components/dot-workflow-actions/dot-workflow-actions.component.html [hunk 1]
```diff
@@ -1,18 +1,20 @@
 @for (action of groupedActions(); track $index) {
-    @if (action.length > 1) {
+    @let mainAction = action.mainAction;
```
Comment (KevinDavilaDotCMS) [inline core-web/libs/ui/src/lib/components/dot-workflow-actions/dot-workflow-actions.component.html:2]: Nice!

**Review by jdotcms — APPROVED** [review]

**Review by KevinDavilaDotCMS — APPROVED** [review]

**Review by KevinDavilaDotCMS — APPROVED** [review]

File: core-web/libs/ui/src/lib/components/dot-workflow-actions/dot-workflow-actions.component.ts [hunk 2]
```diff
@@ -34,71 +39,121 @@ const InplaceButtonSizePrimeNg: Record<ButtonSize, string> = {
...
     @Input() size: ButtonSize = 'normal';
+    /**
+     * Emits when an action is selected
+     *
+     * @memberof DotWorkflowActionsComponent
+     */
     @Output() actionFired = new EventEmitter<DotCMSWorkflowAction>();
 
-    protected groupedActions = signal<MenuItem[][]>([]);
+    protected groupedActions = signal<WorkflowActionsGroup[]>([]);
```
Comment (KevinDavilaDotCMS) [inline core-web/libs/ui/src/lib/components/dot-workflow-actions/dot-workflow-actions.component.ts:75]:
```suggestion
    protected $groupedActions = signal<WorkflowActionsGroup[]>([]);
```

**Review by fmontes — APPROVED** [review]

File: core-web/libs/ui/src/lib/components/dot-workflow-actions/dot-workflow-actions.component.ts (hunk 2 above)
Comment (rjvelazco) [reply core-web/libs/ui/src/lib/components/dot-workflow-actions/dot-workflow-actions.component.ts]: Done!

File: core-web/libs/ui/src/lib/components/dot-workflow-actions/dot-workflow-actions.component.html [hunk 3]
```diff
@@ -1,18 +1,20 @@
-@for (action of groupedActions(); track $index) {
-    @if (action.length > 1) {
+@for (action of $groupedActions(); track $index) {
+    @let mainAction = action.mainAction;
+    @let subActions = action.subActions;
+    @if (subActions.length) {
         <p-splitButton
-            (onClick)="action[0].command({ originalEvent: $event })"
+            (onClick)="mainAction.command({ originalEvent: $event })"
```
Comment (nicobytes) [inline core-web/libs/ui/src/lib/components/dot-workflow-actions/dot-workflow-actions.component.html:6]: According to strict checking the command function could be undefined, I suggest using nullish coalescing like this:

`mainAction?.command({ originalEvent: $event })`

File: core-web/libs/ui/src/lib/components/dot-workflow-actions/dot-workflow-actions.component.ts [hunk 4]
```diff
@@ -34,71 +39,121 @@ const InplaceButtonSizePrimeNg: Record<ButtonSize, string> = {
     changeDetection: ChangeDetectionStrategy.OnPush
 })
 export class DotWorkflowActionsComponent implements OnChanges {
+    /**
+     * List of actions to display
+     *
+     * @type {DotCMSWorkflowAction[]}
+     * @memberof DotWorkflowActionsComponent
+     */
     @Input({ required: true }) actions: DotCMSWorkflowAction[];
```
Comment (nicobytes) [inline core-web/libs/ui/src/lib/components/dot-workflow-actions/dot-workflow-actions.component.ts:48]: I suggest using Signal Input here and in other inputs

File: core-web/libs/ui/src/lib/components/dot-workflow-actions/dot-workflow-actions.component.html (hunk 3 above)
Comment (rjvelazco) [reply core-web/libs/ui/src/lib/components/dot-workflow-actions/dot-workflow-actions.component.html]: Done!

File: core-web/libs/ui/src/lib/components/dot-workflow-actions/dot-workflow-actions.component.ts (hunk 4 above)
Comment (rjvelazco) [reply core-web/libs/ui/src/lib/components/dot-workflow-actions/dot-workflow-actions.component.ts]: Done!

File: core-web/libs/ui/src/lib/components/dot-workflow-actions/dot-workflow-actions.component.html [hunk 5]
```diff
@@ -1,24 +1,26 @@
-@for (action of groupedActions(); track $index) {
-    @if (action.length > 1) {
+@for (action of $groupedActions(); track $index) {
+    @let mainAction = action.mainAction;
+    @let subActions = action.subActions;
```
Comment (oidacra) [inline core-web/libs/ui/src/lib/components/dot-workflow-actions/dot-workflow-actions.component.html:3]: 🤛🏼

File: core-web/libs/ui/src/lib/components/dot-workflow-actions/dot-workflow-actions.component.ts [hunk 6]
```diff
@@ -34,71 +39,121 @@ const InplaceButtonSizePrimeNg: Record<ButtonSize, string> = {
...
+     * @type {ButtonSize}
+     * @memberof DotWorkflowActionsComponent
+     */
+    size = input<ButtonSize>('normal');
+    /**
+     * Emits when an action is selected
+     *
+     * @memberof DotWorkflowActionsComponent
+     */
     @Output() actionFired = new EventEmitter<DotCMSWorkflowAction>();
```
Comment (oidacra) [inline core-web/libs/ui/src/lib/components/dot-workflow-actions/dot-workflow-actions.component.ts:73]: The only one missing to convert to signal.

**Review by oidacra — APPROVED** [review]

**Review by zJaaal — APPROVED** [review]

**Pull request merged**